### PR TITLE
Switch to using AxesGrid internally

### LINKED ISSUE #10: Use AxesGrid internally?
The `AxesGrid` framework offers much of the functionality implemented here and is already in `matplotlib`.   It does not offer an equivalent to the `cbar_short_side_pad` parameter, but past that I think everything implemented here would be possible with `AxesGrid` (when wrapped correctly).  

`AxesGrid` is not perfect, particularly with sharing axes and Cartopy, but it is probably better to work with what `matplotlib` has designed specifically for this purpose than rolling our own solution.

What I would be interested in adding on top of `AxesGrid`:

- ~~The ability to set the colorbar thickness in inches [this is simple; it's just the ratio of the desired colorbar thickness to the figure width or height (depending on its location)]~~ Edit: this is already possible in `AxesGrid` by spacing a `float` value for the `cbar_size` argument.
- The ability to set the figure size in inches
- The ability to set the aspect ratio of each of the panels in the figure
- The ability to set the outer padding of the figure in inches (we can use the `rect` parameter to set the extent of the inner panel objects in the overall figure).  

Choosing the figure size will require some calculation if we want precise control over the aspect ratio of the panels.  While `AxesGrid` has an `aspect` parameter, it only takes values of `True` or `False` (one cannot set a decimal value for instance).  When `True` it just takes the aspect ratio of the axes limits (so if the scale of the axes values differs a lot, this is a problem.  That said, it works pretty well for `cartopy`, because that is what we want in that case).  Ultimately though, I think that is much less work than we currently do here. 

With what we've explored in #9, perhaps it might even be possible to hack together a workaround for the sharing axes issues with cartopy.

As a reference here is a link to the documentation of `ImageGrid`, which is the same as `AxesGrid`: https://matplotlib.org/2.0.2/mpl_toolkits/axes_grid/users/overview.html#axes-grid1

### PR DESCRIPTION
**Pull request by spencerkclark**
Closes #10 

I still need to update the tests and clear out the old implementation, but this is a cleaned up version of the [gist](https://gist.github.com/spencerkclark/7bd3734a6bc7752163b2d0762ded8af7) posted in #10.

### CODE REVIEW

**Comment by spencerkclark** [comment]
@nathanieltarshish I feel like this is far enough along now for you to have a look.  Whenever you get a chance, it would be great if you could review these updates.  This is more or less a total rewrite of `facets.py` and `test_facets.py`, so the diffs look a little more complicated than perhaps they should.  Therefore I might recommend reviewing the raw files from this PR:
- [facets.py](https://github.com/spencerkclark/facets/blob/d6ec64d08a9d3a7f4296eae6fd8f4d2e4c45c605/facets/facets.py)
- [test_facets.py](https://github.com/spencerkclark/facets/blob/d6ec64d08a9d3a7f4296eae6fd8f4d2e4c45c605/facets/tests/test_facets.py)

and then adding comments to the diffs where appropriate.

File: facets/facets.py [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
 
 def facets(rows, cols, width=8., aspect=0.618, top_pad=0.25,
            bottom_pad=0.25, left_pad=0.25, right_pad=0.25, internal_pad=0.33,
-           cbar_mode=None, cbar_short_side_pad=0.5, cbar_long_side_pad=0.5,
-           cbar_thickness=0.125, cbar_location='right', axes_kwargs={}):
+           cbar_mode=None, cbar_short_side_pad=0.5, cbar_pad=0.5,
+           cbar_size=0.125, cbar_location='right', axes_kwargs={}):
     """Create figure and tiled axes objects with precise attributes
 
     Parameters
```
Comment (nathanieltarshish) [inline facets/facets.py]: Need to update documentation for `cbar_mode`

File: facets/facets.py [hunk 2]
```diff
@@ -50,499 +49,147 @@ def facets(rows, cols, width=8., aspect=0.618, top_pad=0.25,
...
-        -------
-        list of Axes objects, list colorbar Axes objects
-        """
-        axes_ = [self.fig.add_axes(tile.transformed_axes_bounds()[0],
-                                   **add_axes_kwargs) for tile in
-                 self.tiles.flatten()]
-        caxes = [self.fig.add_axes(tile.transformed_axes_bounds()[1]) for
-                 tile in self.tiles.flatten()]
-        return axes_, caxes
+    def height(self):
```
Comment (nathanieltarshish) [inline facets/facets.py:137]: Maybe add a comment like "computes the total height of the figure in inches"

**Comment by nathanieltarshish** [comment]
Looks great! 

If you are satisfied, then let's merge into master. Once merged, we can start refining and make the tweaks we discussed today.

**Comment by spencerkclark** [comment]
@nathanieltarshish yes, that sounds like a good plan.  This should be all set now.  Thanks for the review!

**Pull request merged**